docs/Frequently Asked Questions/React Native/index.md

```diff
@@ -0,0 +1,251 @@
+}];
+
+self.sdlManager.screenManager.softButtonObjects = @[softButton];
+```
+
+##### Swift
+```swift
+let softButton = SDLSoftButtonObject(name: "Button", state: SDLSoftButtonState(stateName: "State", text: "Data", artwork: nil), handler: { (buttonPress, butonEvent) in
+    guard buttonPress == nil else { return }
+    
+    let userInfo = ["sdlManager": self.sdlManager]
+    NotificationCenter.default.post(name: NSNotification.Name(rawValue: <#Notification Name#>), object: nil, userInfo: managers)
+})
+
+self.sdlManager.screenManager.softButtonObjects = [softButton];
+```
+
+#### Create the EventEmitter Bridge Class
+
+Create the class that will be the listener for the notiification you created above. This class will be sending and receiving messages from your JavaScript code (React Native). There is a required method you must include in order to be a React Native `EventEmitter`; the method is in the example below. The required `supportedEvents` method returns an array of supported events. Sending an event name that is not included in the array will result in an error. `supportedEvents` tells React Native the events it should observe. An event is sending a message from Native code to React Native code.
+
+##### Objective-C
+###### SDLEventEmitter.h
+```objc
+#import <React/RCTEventEmitter.h>
+#import <React/RCTBridgeModule.h>
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SDLEventEmitter : RCTEventEmitter
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+###### SDLEventEmitter.m
+```objc
+#import "SDLEventEmitter.h"
+#import "ProxyManager.h"
+#import <React/RCTConvert.h>
+#import <SmartDeviceLink/SmartDeviceLink.h>
+
+@implementation SDLEventEmitter
+
+RCT_EXPORT_MODULE()
+
+- (instancetype)init {
+    self = [super init];
+    // Subscribe to event notifications sent from ProxyManager
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(getDoActionNotification:) name:<#Notification Name#> object:nil];
+
+    return self;
+}
+
+// Required Method 
+- (NSArray<NSString *> *)supportedEvents {
+    return @[@"DoAction"];
+}
+
+// Run this code when the subscribed event notification is received
+- (void)getDoActionNotification:(NSNotification *)notification {
+    if(self.sdlManager == nil) {
+        self.sdlManager = notification.userInfo[@"sdlManager"];
+    }
+    [self sendEventWithName:@"DoAction" body:@{@"type": @"actionType"}];
+}
+
+@end
+```
+
+##### Swift
+```swift
+@objc(SDLEventEmitter)
+class SDLEventEmitter: RCTEventEmitter {
+
+    override init() {
+        // Subscribe to event notifications sent from ProxyManager
+        NotificationCenter.default.addObserver(self, selector: #selector(doAction(_:)), name: Notification.Name(rawValue: "<#Notification Name#>", object: nil)
+        super.init()
+    }
+
+    // Run this code when the subscribed event notification is received
+    @objc func doAction(_ notification: Notification) {
```

Comment:
Move this below `supportedEvents` like in the Objective-C code; they should be in the same order.